code/game/turfs/turf.dm

```diff
@@ -343,17 +355,19 @@
 			CHECK_TICK
 
 /turf/narsie_act(force, ignore_mobs, probability = 20)
+	var/list/contents_cached = contents
 	. = (prob(probability) || force)
-	for(var/I in src)
+	for(var/I in contents_cached)
 		var/atom/A = I
 		if(ignore_mobs && ismob(A))
 			continue
 		if(ismob(A) || .)
 			A.narsie_act()
 
-/turf/ratvar_act(force, ignore_mobs, probability = 40)
+/turf/ratvar_act(force, ignore_mobs, probability = 40) // todo: merge with above somehow, copypasta
```

Comment:
eldrich_act with another arg being god?

Reply:
I'd need to refactor stuff, out of scope

Reply:
I was just giving a suggestion based on the comment.